main/src/ca/uwaterloo/flix/language/phase/unification/set/SetSubstitution.scala

```diff
@@ -143,6 +143,14 @@ case class SetSubstitution(m: Map[Int, SetFormula]) {
     }
   }
 
+  /**
+    * Adds the disjoint binding `[x -> f]` to `this`.
+    *
+    * OBS: The [[SetFormula.Var]] of `this`, `x`, and `f` must not overlap.
```

Comment:
This would be costly to check, right?

Reply:
Like with an assertion.

Reply:
yeah I didn't think it was worth it. And technically the mapping `x -> x` is the same as not having `x` so we should also account for that in the check

Reply:
But we could constrain it to its less general form, and check that all mappings x -> f have f.isground and that the keys are disjoint. Also expensive but less cumbersome and its enough for the sve use